text/0052-reserved-namespaces.md

```diff
@@ -0,0 +1,102 @@
+# Reserve Identifiers for Internal Use
+
+## Related issues and PRs
+
+- Reference Issues: (fill in existing related issues, if any)
+- Implementation PR(s): (leave this empty)
+
+## Timeline
+
+- Started: 2024-02-14
+
+## Summary
+
+This RFC extends the reservation of the `__cedar` namespace in schema in [RFC24](https://github.com/cedar-policy/rfcs/blob/main/text/0024-schema-syntax.md) and
+proposes reserving all namespaces that start with two underscores (`__`),
+forbidding their use in Cedar policies, schema, and entity data so that they can be used by future Cedar language features.
+
+## Basic example
+
+Under this RFC the names `__cedar::Long`, `__Long`, and `__namespace::User` will all become invalid in all contexts.
+
+These would be rejected in a policy
+
+```cedar
+permit(principal == __cedar::User::"alice", actions, resource);
+```
+
+in a schema
+
+```
+namespace __DocCloud {
+  ...
+}
+```
+
+and in entity data
+
+```json
+[
+    {
+        "uid": { "__entity": { "type": "__cedar::User", "id": "alice"} },
```

Comment:
Not sure if this was your intention, but you are highlighting that `__entity` is something that starts with two underscores. It is part of entity formatting, not entity data, but it's a little confusing whether you mean that it's ruled out, too. You might point out that `__entity` illustrates the use of `__` as a prefix to avoid conflict with user-defined names, and this is a precedent set by C/C++ (for example): https://devblogs.microsoft.com/oldnewthing/20230109-00/?p=107685